presto-main/src/main/java/com/facebook/presto/metadata/FunctionRegistry.java

```diff
@@ -488,6 +488,9 @@ public Signature resolveFunction(QualifiedName name, List<TypeSignature> paramet
         if (name.getSuffix().startsWith(MAGIC_LITERAL_FUNCTION_PREFIX)) {
             // extract type from function name
             String typeName = name.getSuffix().substring(MAGIC_LITERAL_FUNCTION_PREFIX.length());
+            // The RowType have <> that are not allowed in java variable names, so we replace to {}. We can remove this
+            // hack when the type ROW doesn't have them anymore (be ROW(field type+) instead of ROW<type+>(field+))
+            typeName = typeName.replace('{', '<').replace('}', '>');
```

Comment:
Why do we need this? It shouldn't be necessary, I don't think. In the bytecode generation you should be able to strip the "<" and ">"